File: frontend/ClickAndDrive/src/app/services/chat.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ChatMessage } from './models/chat-message';

@Injectable({
  providedIn: 'root'
})
export class ChatService {
  private apiUrl = 'http://localhost:8080/api/chat';

  constructor(private http: HttpClient) {}

  getHistory(userEmail: string, adminEmail: string): Observable<ChatMessage[]> {
    return this.http.get<ChatMessage[]>(`${this.apiUrl}/${userEmail}/${adminEmail}`);
  }

  sendMessage(dto: ChatMessage): Observable<ChatMessage> {
    return this.http.post<ChatMessage>(`${this.apiUrl}/send`, dto);
  }

  getChatUsers(adminEmail: string): Observable<string[]> {
    return this.http.get<string[]>(`${this.apiUrl}/admin/users/${adminEmail}`);
  }

  markAsSeen(receiverEmail: string, senderEmail: string): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/seen/${receiverEmail}/${senderEmail}`, {});
  }
}

File: mobile-version/Navbar-shortened/app/src/main/java/com/example/clickanddrive/clients/services/ChatService.java
package com.example.clickanddrive.clients.services;

import com.example.clickanddrive.dtosample.ChatMessageDTO;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface ChatService {

    @GET("chat/{userEmail}/{adminEmail}")
    Call<List<ChatMessageDTO>> getHistory(
            @Path("userEmail") String userEmail,
            @Path("adminEmail") String adminEmail
    );

    @POST("chat/send")
    Call<ChatMessageDTO> sendMessage(@Body ChatMessageDTO message);

    @GET("chat/admin/users/{adminEmail}")
    Call<List<String>> getChatUsers(@Path("adminEmail") String adminEmail);

    @PUT("chat/seen/{receiverEmail}/{senderEmail}")
    Call<Void> markAsSeen(
            @Path("receiverEmail") String receiverEmail,
            @Path("senderEmail") String senderEmail
    );
}